### PR TITLE
Don't say that 4.0.0 is the latest release

### PR DESCRIPTION
**Pull request by dschwoerer**
Not adding a version avoids having to patch this website after every release

### CODE REVIEW

File: download/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The latest release is always available from [here][latest].
 
 Previous releases are available:
 
-* The latest release, [version 4.0.0][v4.0.0], was realeased 4th February 2017.
+* [Version 4.0.0][v4.0.0], was realeased 4th February 2017.
```
Comment (bendudson) [inline download/index.md]: Thanks! Good spot. Small typo `realeased` -> `released`

**Comment by dschwoerer** [comment]
I am wondering wheter we should even keep the page?

Should download link directly to https://github.com/boutproject/BOUT-dev/releases ?

**Comment by ZedThree** [comment]
Forgot we even had this page! I could add "update download page" to the release checklist, but something more dynamic might avoid the need for that.

**Comment by ZedThree** [comment]
Dynamically updating the page turned out to be a dead end. Updated with all previous releases (including 4.2.3 which I hope to merge later this evening)

**Pull request merged**